vm/systemSmartContracts/esdt.go

```diff
@@ -1225,6 +1254,33 @@ func (e *esdt) setSpecialRole(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 	return vmcommon.Ok
 }
 
+func checkCorrectAddressForNFTCreateMultiShard(token *ESDTDataV2) error {
```

Comment:
so this function check that we have one but only one role on a particular shard based on the last byte address? Do we need to prevent having 2 addresses with different last byte but on the same shard now (because we only have 3 shards)?

Reply:
It does not check shard actually. Only the last byte. So the addresses should differ on last byte.